packages/perseus/src/widgets/__tests__/explanation.cypress.ts

```diff
@@ -0,0 +1,75 @@
+import renderQuestion from "../../../../../testing/render-question-with-cypress";
+import {cypressTestDependencies} from "../../../../../testing/test-dependencies";
+import * as Dependencies from "../../dependencies";
+import * as Perseus from "../../index";
+import {ipsumExample} from "../__testdata__/explanation.testdata";
+
+describe("Explanation Widget", () => {
+    const postContentLinkText = "the system J-25";
+    const contentLinkOne = "synchronic distortion";
+    const contentLinkTwo = "selective molecular polarization";
+    beforeEach(() => {
+        Dependencies.setDependencies(cypressTestDependencies);
+        Perseus.init({skipMathJax: true});
+    });
+
+    it("prevents interacting with actionable items within content when COLLAPSED", () => {
+        // Arrange
+        renderQuestion(ipsumExample);
+        cy.get('[data-perseus-paragraph-index="0"]')
+            .first()
+            .invoke("attr", "tabindex", "-1")
+            .focus();
+
+        // Act - verify tab order (forwards)
+        cy.realPress("Tab");
+        cy.focused().should("have.text", "Explanation");
+        cy.realPress("Tab");
+        cy.focused().should("have.text", postContentLinkText);
+
+        // Act - verify tab order (backwards)
+        cy.realPress(["Shift", "Tab"]);
+        cy.focused().should("have.text", "Explanation");
+    });
+
+    it("allows interacting with actionable items within content when EXPANDED", () => {
+        // Arrange
+        renderQuestion(ipsumExample);
+        cy.get('[data-perseus-paragraph-index="0"]')
+            .first()
+            .invoke("attr", "tabindex", "-1")
+            .focus();
+        cy.realPress("Tab");
+        cy.focused().should("have.text", "Explanation");
+        cy.realPress("Enter"); // Expand content
+
+        // Act - verify tab order (forwards)
+        cy.realPress("Tab");
+        cy.focused().should("have.text", contentLinkOne);
+        cy.realPress("Tab");
+        cy.focused().should("have.text", contentLinkTwo);
+        cy.realPress("Tab");
+        cy.focused().should("have.text", postContentLinkText);
+
+        // Act - verify tab order (backwards)
+        cy.realPress(["Shift", "Tab"]);
+        cy.focused().should("have.text", contentLinkTwo);
+        cy.realPress(["Shift", "Tab"]);
+        cy.focused().should("have.text", contentLinkOne);
+        cy.realPress(["Shift", "Tab"]);
+        cy.focused().should("have.text", "Hide");
+
+        // Act - confirm collapse
+        cy.realPress("Space"); // Collapse content
+        cy.waitUntil(() =>
+            cy
+                .focused()
+                .siblings()
+                .first()
+                .should("have.attr", "aria-hidden", "true")
+                .should("have.css", "visibility", "hidden"),
+        );
+        cy.realPress("Tab");
+        cy.focused().should("have.text", postContentLinkText);
```

Comment:
It kind of feels like this test is testing too much. We prefer to have each test focus on a single behavior, so that when it fails you can easily tell what's broken.

Since the collapse behavior is covered by unit tests, I don't think we need to test it here. Is there a need to test the backwards tab order? It's the browser default, and we'd have to go out of our way to break it. Testing the forward tab order should be sufficient IMO.

Reply:
I broke out the "verify collapsing still works" portion of the 2nd test to it's own test to reduce the number of behaviors being evaluated in that test.

Call me paranoid, but I've seen situations where event handling works fine for "forward" tabbing, but messes things up when "reverse" tabbing. We don't have event handling happening here, but if we add it in the future, I think it would be good to confirm that we didn't regress. Also, since [LEMS-199](https://khanacademy.atlassian.net/browse/LEMS-199) references moving backwards and having problems (though that is a screen reader action, not a keyboard tab), it was fresh in my mind as something to check.